app/controllers/ChoiceController.scala

```diff
@@ -89,4 +88,23 @@ class ChoiceController @Inject()(
       )
   }
 
+  private def updateChoice(id: String, choice: Choice)(implicit hc: HeaderCarrier) =
+    exportsCacheService.get(id).map(_.map(_.copy(choice = choice.value))).flatMap {
```

Comment:
I'll prefer to create method in cacheService to retrieve this if necessary, `map -> map -> copy -> flatMap` it can be more readable when you will wrap this in a method

Reply:
Not sure about this.  More readable here perhaps, but the service method would(or could) look the same.    Took me ages to get the right combination of maps as well.